### PR TITLE
 Migrate hostmetricsreceiver from Collector core to Collector contrib 

### PR DESCRIPTION
**Pull request by alexperez52**
This PR migrates the hostmetricsreceiver component from the Collector core repository to the Collector contrib repository.

This PR retains the commit history from core and includes plumbing fixes for go.mod, makefile, components.go.

Related issues include: 
[opentelemetry-collector/#3474](https://github.com/open-telemetry/opentelemetry-collector/issues/3474)
[opentelemetry-collector/#3638](https://github.com/open-telemetry/opentelemetry-collector/issues/3638)

### CODE REVIEW

**Comment by alolita** [comment]
Hi @bogdandrutu @tigrannajaryan based on our discussion last week, we're starting to file PRs to move each core component to contrib. Please review and merge. Thanks!

File: cmd/otelcontribcol/components.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filelogreceiver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/fluentforwardreceiver"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver"
```
Comment (mx-psi) [inline cmd/otelcontribcol/components.go:71]: You need to add a `replace` directive on the main `go.mod` file for this to work

File: receiver/hostmetricsreceiver/codegen.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !windows
+
+//go:generate mdatagen metadata.yaml
```
Comment (mx-psi) [inline receiver/hostmetricsreceiver/codegen.go:17]: I don't think we have the `mdatagen` tool on this repository, we will need to move it here first.

File: receiver/hostmetricsreceiver/config_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/configtest"
+	"go.opentelemetry.io/collector/internal/processor/filterset"
```
Comment (mx-psi) [inline receiver/hostmetricsreceiver/config_test.go:28]: This is an internal package from a different repository, so we need to move this over here first or find an alternative solution.

**Review by mx-psi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**